tests/foreman/api/test_smartproxy.py

```diff
@@ -22,282 +22,282 @@
 from robottelo.api.utils import one_to_many_names
 from robottelo.cleanup import capsule_cleanup
 from robottelo.config import settings
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.decorators import skip_if_not_set
 from robottelo.helpers import default_url_on_new_port
 from robottelo.helpers import get_available_capsule_port
-from robottelo.test import APITestCase
 
 
-@pytest.mark.run_in_one_thread
-class CapsuleTestCase(APITestCase):
-    """Tests for Smart Proxy (Capsule) entity."""
+pytestmark = [pytest.mark.run_in_one_thread]
 
-    def _create_smart_proxy(self, **kwargs):
-        """Create a Smart Proxy and register the cleanup function"""
-        proxy = entities.SmartProxy(**kwargs).create()
-        # Add proxy id to cleanup list
-        self.addCleanup(capsule_cleanup, proxy.id)
-        return proxy
 
-    @skip_if_not_set('fake_capsules')
-    @pytest.mark.tier1
-    def test_negative_create_with_url(self):
-        """Proxy creation with random URL
+@pytest.fixture(scope='class')
```

Comment:
what would the correct scope be, now that it isn't used in a class?

Reply:
Thanks, I missed that one! Module scope fits.